### PR TITLE
feat: improve exports and add stackgen main package

### PR DESCRIPTION
**Pull request by Pagebakers**
---

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/StackBakery/stackgen/blob/master/CONTRIBUTING.md)

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license_

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -1725,12 +1725,12 @@ __metadata:
   peerDependencies:
     "@stackgen/core": "workspace:packages/core"
   bin:
-    sg: ./dist/index.js
-    stackgen: ./dist/index.js
+    sg: ././dist/index.js
+    stackgen: ././dist/index.js
```
Comment (moltar) [inline yarn.lock]: this is bad

File: packages/stackgen/package.json [hunk 2]
```diff
@@ -0,0 +1,339 @@
+{
+  "name": "stackgen",
+  "license": "Apache-2.0",
+  "version": "0.0.1",
+  "author": {
+    "name": "Justin McCormick",
+    "email": "me@justinmccormick.com"
+  },
```
Comment (moltar) [inline packages/stackgen/package.json:8]: shall we refactor this? 😁

File: packages/cli/package.json [hunk 3]
```diff
@@ -6,17 +6,17 @@
...
-  "exports": "dist/index.js",
-  "types": "dist/index.d.ts",
+  "main": "./dist/index.js",
+  "exports": "./dist/index.js",
+  "types": "./dist/index.d.ts",
   "bin": {
-    "sg": "./dist/index.js",
-    "stackgen": "./dist/index.js"
+    "sg": "././dist/index.js",
+    "stackgen": "././dist/index.js"
```
Comment (moltar) [inline packages/cli/package.json]: bad

File: .stackgenrc.ts [hunk 4]
```diff
@@ -76,6 +89,8 @@ new YarnProject(workspace, "core", {
   ],
   scripts: {
     yalc: "npx yalc publish",
+    precompile: "yarn clean",
+    compile: `tsup --dts --out-dir ${buildPath} --entry ./index.ts --entry ./util/logger.ts`,
```
Comment (moltar) [inline .stackgenrc.ts:93]: Won't two entries create bloated dist files, as they will bundle all deps twice? Need some sort of vendoring, I think?

**Review by moltar — CHANGES_REQUESTED** [review]

File: .stackgenrc.ts (hunk 4 above)
Comment (Pagebakers) [reply .stackgenrc.ts:93]: I checked, and you are correct.

Probably need a better tsup defineConfig to make sure that doesn't happen. but this fixes the build for now.

File: packages/cli/package.json (hunk 3 above)
Comment (Pagebakers) [reply packages/cli/package.json]: Oops

File: packages/stackgen/package.json (hunk 2 above)
Comment (Pagebakers) [reply packages/stackgen/package.json:8]: yeah let's do it in another PR. We need to setup a name / email address.

File: .stackgenrc.ts (hunk 4 above)
Comment (moltar) [reply .stackgenrc.ts:93]: Ok, I will resolve, but please backlog this so we don't forget.

**Review by moltar — APPROVED** [review]

**Pull request merged**